### DISCUSSION TITLE
Can I get the number of users participating in the realtime subscription?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#3587] [Questions] [answered]
I want to get the number of users participating in realtime subscription, is it possible?
I would like to get the number of people participating in a chat session.

**Comment by dshukertjr** [comment, marked as answer]
Hi @ru88s!

You can achieve this by having a `participants` table. In that table, you will have a `last_online_at` column where you will hold when each user was present in each room at what time. Your users will create a row in this table when they join a room with their `user_id`, `room_id` of the chat room they joined, and the current timestamp. From there on, they will update the row every minute or so with the current timestamp. 

| user_id | room_id | last_online_at            |   
|---------|---------|---------------------------|
| aaa       | 12      | 2021-10-19T23:09:02+00:00 |   
| bbb       | 12      | 2021-10-19T23:09:03+00:00 |   
| ccc       | 24      | 2021-10-19T23:08:02+00:00 |   

With this, you can setup a listener for this table to get everyone in a chat room. You can loop through them and get the number of users who have been online in the past minute to get the number of participants!

Note that by reducing the recurring update duration from minute to something smaller, you can make this more "realtime".

**Reply by ru88s** [reply]
Hi @dshukertjr! 

You are a genius.
I was able to implement it perfectly! Thank you very much.
